resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ClientRequestHeaders.java

```diff
@@ -293,9 +308,13 @@ public MediaType getMediaType()
          }
          else
          {
-            String str = configuration.toHeaderString(obj);
-            Cookie cookie = Cookie.valueOf(str);
-            cookies.put(cookie.getName(), cookie);
+            String[] cs = ((String) obj).split(";");
+            for (String c : cs)
+            {
+               String str = configuration.toHeaderString(c);

```

Comment:
Why use `toHeaderString` on a string? Should be used on the input instead of the cast to string IMO.